pallets/asset-index/src/lib.rs

```diff
@@ -1034,8 +1035,8 @@ pub mod pallet {
 			}
 			// native asset can't be added
 			Self::ensure_not_native_asset(&asset_id)?;
-			// mint asset into the treasury account
-			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+			// transfer the asset from the caller to treasury account
+			T::Currency::transfer(asset_id, caller, &Self::treasury_account(), units)?;
```

Comment:
the caller account may not have enough balance to make transfer

Reply:
yeah but we catch the error and abort in case that happens?

Reply:
> The given amount of assets must already exist in the caller's account

After modifying this to `transfer`, we don't have any extrinsic left can mint assets in our pallets, for approving an asset working in PINT, we need to mint the assets with orml pallet by hand, or we'll mint them from the XCM part?

Reply:
this is intended as discussed here https://github.com/ChainSafe/PINT/pull/401#discussion_r712791918 
Minting liquid assets without receiving them via XCM is a soundness issue